### PR TITLE
Add check for job.Filter before validation

### LINKED ISSUE #1703: Logpush null filter results in unexpected end of JSON input
### Confirmation

- [X] My issue isn't already found on the issue tracker.
- [X] I have replicated my issue using the latest version of the provider and it is still present.

### Terraform and Cloudflare provider version

terraform: 1.1.7
cloudflare: 3.17.0 (was using 3.16.0 to create the existing logpush job)

### Affected resource(s)

resource cloudflare_logpush_job

### Terraform configuration files

```hcl
resource "cloudflare_logpush_job" "example_job" {
  enabled             = true
  zone_id             = "..."
  name                = "..."
  logpull_options     = "fields=..."
  destination_conf    = "..."
  ownership_challenge = "..."
  dataset             = "..."
}
```

### Debug output

Can provide if necessary - started to run but way too many lines of logs that it broke build :( 

### Panic output

_No response_

### Expected output

Expected filter = optional, logpush job creation successful

### Actual output

For apply/destroy: 
```
Error: error parsing logpush job from resource: unexpected end of JSON input
```

### Steps to reproduce

1. Create new logpush job without filter attribute -> will fail? Haven't tested yet
2. Update existing logpush job -> will fail, error message, adds filter = null to state
3. Try destroying existing logpush job -> will fail, error message

### Additional factoids

_No response_

### References

_No response_

### PR DESCRIPTION
**Pull request by sawp-d**
Related to issue #1703, hoping this will fix the error?

Closes #1703

### CODE REVIEW

**Comment by jacobbednarz** [comment]
changelog detected :white_check_mark:

File: .changelog/1704.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:bug
+resource/cloudflare_logpush_job: Fix null 'filter' attribute throwing "error parsing logpush job from resource: unexpected end of JSON input"
```
Comment (jacobbednarz) [inline .changelog/1704.txt:2]:
```suggestion
resource/cloudflare_logpush_job: fix null `filter` attribute throwing parsing error
```

**Comment by sawp-d** [comment]
Hey @jacobbednarz! Do you know why the changelog build may be failing? Looks like that’s the last! 😕

**Comment by jacobbednarz** [comment]
i'm afriad the proposed fix here won't actually achieve what you're trying to do as the underlying struct will still be present. cloudflare/cloudflare-go#937 is the fix for this.

**Comment by sawp-d** [comment]
No worries, glad it was fixed in cloudflare-go!

**Pull request closed without merging**